RecoLocalCalo/HGCalRecAlgos/interface/HGCalMultiCluster.h

```diff
@@ -0,0 +1,93 @@
+#ifndef RecoLocalCalo_HGCalRecAlgos_HGCalMultiCluster
+#define RecoLocalCalo_HGCalRecAlgos_HGCalMultiCluster
+
+#include <vector>
+
+#include "DataFormats/EgammaReco/interface/BasicCluster.h"
+
+
+class HGCalMultiCluster{
+
+public:
+  typedef std::vector<reco::BasicCluster>::const_iterator component_iterator;
+  typedef std::vector<reco::BasicCluster> ClusterCollection;
+
+  HGCalMultiCluster(){
+  }
+  HGCalMultiCluster(ClusterCollection &thecls) : myclusters(thecls){
+  }
+  HGCalMultiCluster(const reco::BasicCluster &thecl) :  myclusters(1,thecl){
+  }
+  void push_back(const reco::BasicCluster &b){
+    myclusters.push_back(b);
+  }
+  unsigned int size()
+  {
+    return myclusters.size();
+  }						
+  component_iterator begin() const {
+    return myclusters.begin();
+  }
+  component_iterator end() const {
+    return myclusters.end();
+  }
+  double simple_z(double vz) const {
+    double acc = 0.0;
+    double totweight = 0.;
+    for(component_iterator it = begin(); it != end(); it++){
+      acc += (it->z()-vz)*it->energy()*it->size();
+      totweight+=it->energy()*it->size();
+    }
+    return acc/totweight;
+  }
+
+  double simple_slope_x(double vz) const {
+    double acc = 0.0;
+    double totweight = 0.;
+    for(component_iterator it = begin(); it != end(); it++){
+      acc += it->x()/(it->z()-vz)*it->energy()*it->size();
+      totweight+=it->energy()*it->size();
+    }
+    return acc/totweight;
+  }
+  double simple_slope_y(double vz) const {
+    double acc = 0.0;
+    double totweight = 0.;
+    for(component_iterator it = begin(); it != end(); it++){
+      acc += it->y()/(it->z()-vz)*it->energy()*it->size();
+      totweight+=it->energy()*it->size();
+    }
+    return acc/totweight;
+  }
```

Comment:
The proposed change is elegant but it will end up having about as many or more lines of code and moves the problem of maintaining different but similar functionality from the methods to the template argument FUNC... so in my opinion it is not an improvement...
I agree to leave this discussion for later